src/System.IO/src/System/IO/StreamWriter.cs

```diff
@@ -66,22 +66,7 @@ private void CheckAsyncTaskInProgress()
         // Even Close() will hit the exception as it would try to flush the unwritten data. 
         // Maybe we can add a DiscardBufferedData() method to get out of such situation (like 
         // StreamReader though for different reason). Either way, the buffered data will be lost!
-        private static volatile Encoding s_utf8NoBOM;
-
-        internal static Encoding UTF8NoBOM
-        {
-            //[FriendAccessAllowed]
-            get
-            {
-                if (s_utf8NoBOM == null)
-                {
-                    // No need for double lock - we just want to avoid extra
-                    // allocations in the common case.
-                    s_utf8NoBOM = new UTF8Encoding(false, true);
-                }
-                return s_utf8NoBOM;
-            }
-        }
+        private static Encoding UTF8NoBOM => EncodingCache.UTF8NoBOM;
```

Comment:
I was going to get rid of this static property on `StreamWriter`, but then I didn't have a great place to move the above comment. Also, I'm planning to submit a similar change to CoreCLR repo, where it's better to keep this property because it has `[FriendAccessAllowed]`, and this keeps the diff minimal between the repos.